test/common_e2e.dart

```diff
@@ -35,62 +28,22 @@ const String DEFAULT_KEY_LOCATION =
 // attempt to account for that.
 const STORAGE_LIST_DELAY = Duration(seconds: 5);
 
-bool onBot() {
-  // When running on the package-bot the current user is chrome-bot.
-  var envName;
-  if (Platform.isWindows) {
-    envName = 'USERNAME';
-  } else {
-    envName = 'USER';
-  }
-  return Platform.environment[envName] == 'chrome-bot';
-}
-
-// Get the service key from the specified location.
-Future<String> serviceKeyJson(String serviceKeyLocation) {
-  if (!serviceKeyLocation.startsWith('gs://')) {
-    return File(serviceKeyLocation).readAsString();
-  }
-  Future<ProcessResult> future;
-  if (onBot()) {
-    future = Process.run(
-        'python', ['third_party/gsutil/gsutil', 'cat', serviceKeyLocation],
-        runInShell: true);
-  } else {
-    var gsutil = Platform.isWindows ? 'gsutil.cmd' : 'gsutil';
-    future = Process.run(gsutil, ['cat', serviceKeyLocation]);
-  }
-  return future.then((result) {
-    if (result.exitCode != 0) {
-      throw Exception('Failed to run gsutil, ${result.stderr}');
-    }
-    return result.stdout.toString();
-  });
-}
-
-typedef AuthCallback = Future Function(String project, http.Client client);
-
-Future withAuthClient(List<String> scopes, AuthCallback callback,
-    {bool trace = false}) {
+Future<T> withAuthClient<T>(
+  List<String> scopes,
+  Future<T> Function(String project, http.Client client) callback, {
+  bool trace = false,
+}) async {
   var project = Platform.environment[PROJECT_ENV];
-  var serviceKeyLocation = Platform.environment[SERVICE_KEY_LOCATION_ENV];
 
-  if (!onBot() && (project == null || serviceKeyLocation == null)) {
-    throw StateError(
-        'Environment variables $PROJECT_ENV and $SERVICE_KEY_LOCATION_ENV '
-        'required when not running on the package bot');
+  if (project == null) {
+    throw StateError('Environment variables $PROJECT_ENV ');
   }
 
-  project = project ?? DEFAULT_PROJECT;
-  serviceKeyLocation = serviceKeyLocation ?? DEFAULT_KEY_LOCATION;
-
-  return serviceKeyJson(serviceKeyLocation).then((keyJson) {
-    var creds = auth.ServiceAccountCredentials.fromJson(keyJson);
-    return auth
-        .clientViaServiceAccount(creds, scopes)
-        .then((http.Client client) {
-      if (trace) client = TraceClient(client);
-      return callback(project, client);
-    });
-  });
+  http.Client client = await auth.clientViaApplicationDefaultCredentials(
+    scopes: scopes,
+  );
+  if (trace) {
+    client = TraceClient(client);
+  }
+  return await callback(project, client);
```

Comment:
Does return await really help here, by giving a better stack trace in some way?  Otherwise, it is equivalent to return.